### PR TITLE
feat: initial linked keywords additions

### LINKED ISSUE #177: Allow derived/related keywords
If you have the keyword "collaborative", it should be possible to also give out pegs by using derived or related keywords like "collaboration" and "collaborate", while still being tied to the same keyword "collaborative".

Proposed implementation: have a mapping from keywordClass to keywordStems. The class is what would be displayed publicly, while the stem(s) would be what is actually used for matching. For the "collaborative" keywordClass, a single keywordStem of "collaborat" might be sufficient.

Brought forward from https://github.com/GlobalX/pockybot/issues/21

### PR DESCRIPTION
**Pull request by ThomasGillin**
resolves #177

### CODE REVIEW

**Review by Evangelinexx — APPROVED** [review]

**Review by jimcullenaus — CHANGES_REQUESTED** [review]
If I'm understanding what this is doing correctly, it looks pretty good! Just need to add some documentation explaining how it works in the README, and update the Help command to support it.

File: src/PockyBot.NET/Services/Pegs/PegRequestValidator.cs [hunk 1]
```diff
@@ -69,10 +69,14 @@ private void ValidateKeywords(string comment)
         {
             var keywords = _configRepository.GetStringConfig("keyword");
             var penaltyKeywords = _configRepository.GetStringConfig("penaltyKeyword");
+            var linkedKeywords = _configRepository.GetStringConfig("linkedKeyword").Select(x => x.Split(':'));
```
[the remark this answers is not in the document]
Comment (ThomasGillin) [reply src/PockyBot.NET/Services/Pegs/PegRequestValidator.cs:72]: Yes. That's correct. I'll see if I can figure a way to simplify it.

**Review by Chromadream — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -164,8 +164,14 @@ Table `generalconfig` is initalised with default values as follows:
...
-Default keywords are 'customer', 'brave', 'awesome', 'collaborative', and
-'real', shorthands for the GlobalX company values.
+Default keywords are 'customer', 'collaboration', 'amazing', 'integrity', and
+'improvement', shorthands for the Dye & Durham company values.
+
+Table `stringconfig` is also used to define linked keywords.
+Name field is 'linkedKeyword' and 'value' is the value of the keyword and a
+linked word, separated by a colon (e.g. 'amazing:awesome').
+These are used to define other words that will act as one of the main
+keywords for validation and results purposes.
```
Comment (jimcullenaus) [inline README.md:174]: nice!

File: src/PockyBot.NET/Services/Triggers/Help.cs [hunk 3]
```diff
@@ -221,6 +221,8 @@ private string CreateStringConfigHelpMessage(PockyUser user)
             if (HasPermission(user, new[] {Role.Admin, Role.Config})) {
                 return "### How to configure string config values 🎻!\n" +
                     $"1. To get/add/delete string config values, type `@{_pockyBotSettings.BotName} {Commands.StringConfig} {string.Join("|",ConfigActions.All())} {{name}} {{value}}`\n" +
+                    $"    * Example 1: To add a keyword called \"amazing\", type `@{_pockyBotSettings.BotName} {Commands.StringConfig} {ConfigActions.Add} keyword amazing`\n" +
+                    $"    * Example 2: To add a linked keyword called \"awesome\" to the \"amazing\" keyword, type `@{_pockyBotSettings.BotName} {Commands.StringConfig} {ConfigActions.Add} linkedKeyword amazing:awesome`\n" +
```
Comment (jimcullenaus) [inline src/PockyBot.NET/Services/Triggers/Help.cs:225]: looks good!

**Review by jimcullenaus — APPROVED** [review]

**Review by AUSBird — APPROVED** [review]

**Review by jimcullenaus — APPROVED** [review]

**Review by Chromadream — APPROVED** [review]

**Review by Lauraducky — APPROVED** [review]

**Pull request merged**